### PR TITLE
Update fog to 1.42

### PR DESCRIPTION
**Pull request by xprazak2**
For plugin updates, please indicate which repos this should be built into:

* [x] Nightly
* [ ] 1.16
* [ ] 1.15
* [ ] 1.14

See Foreman's [plugin maintainer documentation](http://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin#Release-strategies) for more information.

---

### CODE REVIEW

**Comment by ekohl** [comment]
Related rubygem-fog-ovirt PR at https://github.com/theforeman/foreman-packaging/pull/1970

File: rubygem-fog/rubygem-fog.spec [hunk 1]
```diff
@@ -49,8 +49,7 @@ Requires: %{?scl_prefix}rubygem(fog-xml) >= 0.1.1
 Requires: %{?scl_prefix}rubygem(fog-xml) < 0.2.0
 Requires: %{?scl_prefix}rubygem(ipaddress) >= 0.5
 Requires: %{?scl_prefix}rubygem(ipaddress) < 1.0
-Requires: %{?scl_prefix_ruby}rubygem(json) >= 1.8
-Requires: %{?scl_prefix_ruby}rubygem(json) < 2.0
+Requires: %{?scl_prefix_ruby}rubygem(json) >= 2.0
```
Comment (ekohl) [inline rubygem-fog/rubygem-fog.spec]: This requires us to bump to the newer ruby SCL. Will require some coordination with @ehelms.

**Comment by xprazak2** [comment]
Rebased onto latest rpm/develop, lets see if it turns green.

**Comment by ekohl** [comment]
This is pending a mass rebuild of our existing gems in the new SCL.

**Comment by ehelms** [comment]
[test]

**Comment by ehelms** [comment]
Locally, I had to make this additional change:

```
@@ -42,6 +42,7 @@ Requires: %{?scl_prefix}rubygem(fog-digitalocean) >= 0.3.0
 Requires: %{?scl_prefix}rubygem(fog-google) <= 0.1.0
 Requires: %{?scl_prefix}rubygem(fog-json)
 Requires: %{?scl_prefix}rubygem(fog-openstack)
+Requires: %{?scl_prefix}rubygem(fog-ovirt)
 Requires: %{?scl_prefix}rubygem(fog-rackspace)
 Requires: %{?scl_prefix}rubygem(fog-vsphere) >= 0.4.0
 Requires: %{?scl_prefix}rubygem(fog-xenserver)
@@ -61,6 +62,7 @@ BuildRequires: %{?scl_prefix}rubygem(fog-digitalocean) >= 0.3.0
 BuildRequires: %{?scl_prefix}rubygem(fog-google) <= 0.1.0
 BuildRequires: %{?scl_prefix}rubygem(fog-json)
 BuildRequires: %{?scl_prefix}rubygem(fog-openstack)
+BuildRequires: %{?scl_prefix}rubygem(fog-ovirt)
 BuildRequires: %{?scl_prefix}rubygem(fog-rackspace)
 BuildRequires: %{?scl_prefix}rubygem(fog-vsphere) >= 0.4.0
 BuildRequires: %{?scl_prefix}rubygem(fog-xenserver)
```

**Comment by xprazak2** [comment]
I added fog-ovirt into requirements

**Comment by ekohl** [comment]
Looks like rbovirt needs to be rebuilt into the new SCL first. I'll get on that.

**Comment by ekohl** [comment]
Looks like that's part of https://github.com/theforeman/foreman-packaging/pull/2037

**Comment by ehelms** [comment]
[test]

**Comment by ekohl** [comment]
[test]

**Comment by ekohl** [comment]
Given @dLobatog wanted to wait for the new SCL (which is here now) I'm merging this.

**Pull request merged**